### PR TITLE
[4.0] [plg_privacy_message] Convert to prepared statement

### PR DESCRIPTION
**Pull request by HLeithner**
# Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries. This PR needs https://github.com/joomla-framework/database/pull/162

### Testing Instructions
Use the plugin in all ways you can think of.

Create users
Generate privacy message entries.

### Expected result
Nothing changed.

### CODE REVIEW

**Comment by brianteeman** [comment]
Looks like there are two unrelated files here

**Comment by HLeithner** [comment]
thx, I will fix all 3 PR later

File: plugins/privacy/message/message.php [hunk 1]
```diff
@@ -40,9 +42,11 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
 		$query = $this->db->getQuery(true)
```
Comment (Quy) [inline plugins/privacy/message/message.php]: Do you want to do this here `$db = $this->db;`?

**Comment by HLeithner** [comment]
Removed the orWhere function based discussion https://github.com/joomla-framework/database/issues/174

File: plugins/privacy/message/message.php [hunk 2]
```diff
@@ -36,15 +38,17 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
 			->select('*')
-			->from($this->db->quoteName('#__messages'))
-			->where($this->db->quoteName('user_id_from') . ' = ' . (int) $user->id)
-			->orWhere($this->db->quoteName('user_id_to') . ' = ' . (int) $user->id)
-			->order($this->db->quoteName('date_time') . ' ASC');
+			->from($db->quoteName('#__messages'))
+			->where($db->quoteName('user_id_from') . ' = :useridfrom')
+			->extendWhere('OR', $db->quoteName('user_id_to') . ' = :useridto')
+			->order($db->quoteName('date_time') . ' ASC')
+			->bind(['useridfrom', 'useridto'], $user->id, ParameterType::INTEGER);
```
Comment (SharkyKZ) [inline plugins/privacy/message/message.php]:
```suggestion
			->bind([':useridfrom', ':useridto'], $user->id, ParameterType::INTEGER);
```

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 427423bcbb416f2274fc41cb3b566a74ba8ec531<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25431">issues.joomla.org/tracker/joomla-cms/25431</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 427423bcbb416f2274fc41cb3b566a74ba8ec531<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25431">issues.joomla.org/tracker/joomla-cms/25431</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25431">issues.joomla.org/tracker/joomla-cms/25431</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**